pipelinewise/fastsync/commons/target_snowflake.py

```diff
@@ -31,9 +32,17 @@ class FastSyncTargetSnowflake:
     def __init__(self, connection_config, transformation_config=None):
         self.connection_config = connection_config
         self.transformation_config = transformation_config
-        self.s3 = boto3.client('s3',
-                               aws_access_key_id=self.connection_config['aws_access_key_id'],
-                               aws_secret_access_key=self.connection_config['aws_secret_access_key'])
+        if 'aws_access_key_id' in connection_config and 'aws_secret_access_key' in connection_config:
+            self.use_iam_accesskeys = True
+        else:
+            self.use_iam_accesskeys = False
+
+        if self.use_iam_accesskeys:
+            self.s3 = boto3.client('s3',
+                                   aws_access_key_id=self.connection_config['aws_access_key_id'],
+                                   aws_secret_access_key=self.connection_config['aws_secret_access_key'])
+        else:
+            self.s3 = boto3.client('s3')
```

Comment:
I've reviewed it again and compared it to pipelinewise-target-snowflake [here](https://github.com/transferwise/pipelinewise-target-snowflake/blob/a26fa575cd24bdf63603d845988db90fbc0c5229/target_snowflake/db_sync.py#L291) , where secrets, env vars, iam roles and aws_session_tokens are all supported. 

Can you please make the last change and modify it to be this code in sync with pipelinewise-target-snowflake? The code would be shorter and we'd add support for `aws_session_tokens`as well. I think it'd something like this:

```
        self.s3 = boto3.client(
            's3',
            aws_access_key_id=self.connection_config.get('aws_access_key_id'),
            aws_secret_access_key=self.connection_config.get('aws_secret_access_key'),
            aws_session_token=self.connection_config.get('aws_session_token')
        )
```

Reply:
True it makes sense to initialise the s3 client without any conditions. The condition is also used while loading the data but if you think we could just pass the external stage assuming the credentials for reading from s3 is configured with the external stage, we can get rid of these conditions completely. I will wait for your comment (below) before removing it . Thanks for reviewing it .

Reply:
> we could just pass the external stage assuming the credentials for reading from s3 is configured with the external stage
Yes, this is the correct behaviour. Please see more detailed answer below in the other comment.